utils/extremum_tracking.hh

```diff
@@ -10,41 +10,33 @@
 
 #include <functional>
 #include <limits>
+#include <optional>
 
 namespace detail {
 
 template<typename T, typename Comparator>
 class extremum_tracker {
     T _default_value;
-    bool _is_set = false;
-    T _value;
+    std::optional<T> _value;
```

Comment:
@cmm the use case for the above min_max_tracker is anecdotal.
I used it to generate guaranteed-to-be-unique names for the pending_delete log files in `delete_atomically`.
Using the generations range in the log file name is not really needed by anything else, so it can be safely changed to using a (time)uuid for the log name uniquifier instead.  And if we add a suitable printer for it that will also sort lexicographically and temporally the same way - then it would be even nicer.

Reply:
It would be nice to have the lexicographical representation for time_uuid also for incorporating those as the sstable generation as part of https://github.com/scylladb/scylla/issues/10459